src/components/KTableView/KTableView.vue

```diff
@@ -0,0 +1,896 @@
+            >
+              <slot name="empty-state-action-icon" />
+              {{ emptyStateActionMessage }}
+            </KButton>
+          </template>
+        </KEmptyState>
+      </slot>
+    </div>
+
+    <div v-else>
+      <div
+        class="table-wrapper"
+        :style="tableWrapperStyles"
+        @scroll.passive="scrollHandler"
+      >
+        <table
+          v-bind-once="{ 'data-tableid': tableId }"
+          class="table"
+          :class="{
+            'has-hover': rowHover && !isActionsDropdownHovered,
+            'is-clickable': isClickable
+          }"
+        >
+          <thead :class="{ 'is-scrolled': isScrolled }">
+            <tr
+              ref="headerRow"
+              :class="{ 'is-scrolled': isScrolled }"
+            >
+              <th
+                v-for="(column, index) in visibleHeaders"
+                :key="`table-${tableId}-headers-${index}`"
+                :aria-sort="column.key === sortColumnKey ? (sortColumnOrder === 'asc' ? 'ascending' : 'descending') : undefined"
+                class="table-headers"
+                :class="getHeaderClasses(column, index)"
+                :data-testid="`table-header-${column.key}`"
+                :style="columnStyles[column.key]"
+                @click="() => {
+                  if (column.sortable) {
+                    $emit('sort', {
+                      prevKey: sortColumnKey,
+                      sortColumnKey: column.key,
+                      sortColumnOrder: sortColumnOrder === 'asc' ? 'desc' : 'asc' // display opposite because sortColumnOrder outdated
+                    })
+                    sortClickHandler(column)
+                  }
+                }"
+                @mouseleave="currentHoveredColumn = ''"
+                @mouseover="currentHoveredColumn = column.key"
+              >
+                <div
+                  v-if="resizeColumns && index !== 0"
+                  class="resize-handle previous"
+                  @click.stop
+                  @mousedown="startResize($event, visibleHeaders[index - 1].key)"
+                  @mouseleave="resizerHoveredColumn = ''"
+                  @mouseover="resizerHoveredColumn = visibleHeaders[index - 1].key"
+                />
+
+                <div
+                  :aria-describedby="column.tooltip || $slots[getColumnTooltipSlotName(column.key)] ? `${getColumnTooltipSlotName(column.key)}-${tableId}` : undefined"
+                  class="table-headers-container"
+                  :class="{ 'resized': resizingColumn === column.key }"
+                >
+                  <slot
+                    :column="getGeneric(column)"
+                    :name="getColumnSlotName(column.key)"
+                  >
+                    <span
+                      class="table-header-label"
+                      :class="{
+                        'sr-only': column.hideLabel || (column.key === TableViewHeaderKeys.ACTIONS && column.hideLabel !== false),
```

Comment:
since "actions" is special, can we just check for it in the component (you can define the const in here) rather than importing `TableViewHeaderKeys`?

The other places it's used won't evaluate correctly for types as-is